src/mono/mono/metadata/icall.c

```diff
@@ -1293,6 +1293,27 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_RunModuleConstructor (M
 	mono_runtime_class_init_full (vtable, error);
 }
 
+#ifdef ENABLE_NETCORE
+MonoBoolean
+ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_SufficientExecutionStack (void)
+{
+	MonoInternalThread *thread = mono_thread_internal_current ();
+	void *current = &thread;
+
+	// Stack upper/lower bound should have been calculated and set as part of register_thread.
+	// If not, we are optimistic and assume there is enough room.
+	if (!thread->thread_info->stack_start_limit || !thread->thread_info->stack_end)
+		return TRUE;
+
+	// Stack start limit is stack lower bound. Make sure there is enough room left.
+	void *limit = ((uint8_t *)thread->thread_info->stack_start_limit) + MONO_MIN_EXECUTION_STACK_SIZE;
```

Comment:
Instead of `thread->thread_info` I think you can just have
```
MonoThreadInfo *info = mono_thread_info_current();

use (info->stack_start_limit, etc)
```